lib/study.py

```diff
@@ -0,0 +1,248 @@
+            if 'clean' in args:
+                clean = args['clean']
+            for i in self.mad6t_joblist:
+                print("The job %s is running...."%i)
+                mad6t_oneturn.run(i)
+            print("All jobs are completed normally!")
+            os.chdir(cur_path)
+            if clean:
+                shutil.rmtree(execution_field)
+        elif platform.lower() == 'htcondor':
+            #app_path = os.path.abspath(inspect.getfile(self.__class__))
+            #app_path = os.path.dirname(app_path)
+            #sys.path.append(app_path)
+            pass
+        else:
+            print("Invlid platfrom!")
+
+    def prepare_madx_single_input(self):
+        '''Prepare the input files for madx and one turn sixtrack job'''
+        self.config['madx'] = {}
+        madx_sec = self.config['madx']
+        madx_sec['source_path'] = self.paths['location']
+        madx_sec['madx_exe'] = self.paths['madx_exe']
+        madx_sec['mask_name'] = self.madx_input["mask_name"]
+        madx_sec['output_files'] = utils.code(self.madx_output)
+
+        self.config['mask'] = {}
+        mask_sec = self.config['mask']
+
+        self.config['sixtrack'] = {}
+        six_sec = self.config['sixtrack']
+        six_sec['source_path'] = self.paths['location']
+        six_sec['sixtrack_exe'] = self.paths['sixtrack_exe']
+        six_sec['temp_files'] = utils.code(self.oneturn_sixtrack_input['temp'])
+        six_sec['input_files'] = utils.code(self.oneturn_sixtrack_input['input'])
+        six_sec['output_files'] = utils.code(self.oneturn_sixtrack_output)
+        self.config['fort3'] = self.oneturn_sixtrack_params
+
+        keys = sorted(self.madx_params.keys())
+        values = []
+        for key in keys:
+            values.append(self.madx_params[key])
+
+        for element in itertools.product(*values):
+            for i in range(len(element)):
+                ky = keys[i]
+                vl = element[i]
+                mask_sec[ky] = str(vl)
+            input_name = self.madx_name_conven('mad6t', keys, element, '.ini')
+            prefix = self.madx_input['mask_name'].split('.')[0]
+            madx_input_name = self.madx_name_conven(prefix, keys, element)
+            madx_sec['input_name'] = madx_input_name
+            out_name = self.madx_name_conven('Result', keys, element, '')
+            mad6t_input = os.path.join(self.location, 'mad6t_input')
+            madx_sec['dest_path'] = os.path.join(self.paths['madx_out'], out_name)
+            six_sec['dest_path'] = os.path.join(self.paths['madx_out'], out_name)
+            output = os.path.join(mad6t_input, input_name)
+            with open(output, 'w') as f_out:
+                self.config.write(f_out)
+            self.mad6t_joblist.append(output)
+            print('Successfully generate input file %s'%output)
+
+    def madx_name_conven(self, prefix, keys, values, suffix = '.madx'):
+        '''The convention for naming input file'''
+        mk = prefix
```

Comment:
as for other places of the code, I would have the output string in the interface of the method, and return true or false as error handling. Moreover, I would check `len(keys)!=len(values)` and behave accordingly.

Reply:
Same reason, if we put the ```output``` in the interface of the method, I'm afraid it will not work normally.

Reply:
> Same reason, if we put the output in the interface of the method, I'm afraid it will not work  normally.

why?